src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CacheWerk\Relay\Psr\Tracing;
+
+use Relay\Relay;
+
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\TracerInterface;
+use OpenTelemetry\API\Trace\TracerProviderInterface;
+
+class RelayOpenTelemetry
+{
+    /**
+     * The Relay instance.
+     *
+     * @var \Relay\Relay
+     */
+    protected Relay $relay;
+
+    /**
+     * The OpenTelemetry tracer provider instance.
+     *
+     * @var \OpenTelemetry\API\Trace\TracerInterface
+     */
+    protected TracerInterface $tracer;
+
+    /**
+     * Creates a new `NewRelicRelay` instance.
+     *
+     * @param  \Relay\Relay  $relay
+     * @param  \OpenTelemetry\API\Trace\TracerProviderInterface  $tracerProvider
+     * @return void
+     */
+    public function __construct(Relay $relay, TracerProviderInterface $tracerProvider)
+    {
+        $this->relay = $relay;
+        $this->tracer = $tracerProvider->getTracer('Relay', (string) phpversion('relay'));
```

Comment:
>I would recomment to accept a TracerProvider instead of a TracerInterface, because it is up to an instrumentation how to name and version tracers.

@vmihailenco: We now accept a `TracerProviderInterface` and store the `TracerInterface`.

Reply:
> In future, even the `$tracerProvider` might be optional because there likely will be a global/default tracer provider.

@vmihailenco: I read this somewhere. Is this possible already to access the global/default tracer provider.

Reply:
I don't think it is already available, but in other languages it is a common practice to accept a `TracerProvider`, for example, in uptrace-go we have `WithTracerProvider($provider)`.

If you need to make `$tracer` global, you probably should create some kind of global variable in the code for now.

Reply:
I'll see what their team has to say: https://github.com/open-telemetry/opentelemetry-php/discussions/851